slippi-stats.js

```diff
@@ -71,250 +76,311 @@ var final_player_name = user_player
         }
 
-        player_final_percent = game.getLatestFrame().players[player_num].post.percent
-        opponent_final_percent = game.getLatestFrame().players[opponent_num].post.percent
-        end_more_kills = player_kills > opponent_kills
-        end_lower_percent = (player_kills == opponent_kills) && player_final_percent < opponent_final_percent
-
-        // disabled due to perceived inconsistencies
-        // try {
-        //     end_opponent_LRAS = (game.getGameEnd().lrasInitiatorIndex == opponent_num)
-        //     end_player_LRAS = (game.getGameEnd().lrasInitiatorIndex == player_num)
-        // }
-        // catch {
-        //     end_opponent_LRAS = false
-        //     end_player_LRAS = false
-        // } 
-
-        // Every death is considered the opponent's kill
-        // If the player didn't quit out AND has more kills than the opponent, the same but with a lower percent, or the opponent quits out: it's a win, otherwise it's a loss. Ties handled above
-        // if (!end_player_LRAS && (end_more_kills || end_lower_percent || end_opponent_LRAS)) {
-        if (end_more_kills || end_lower_percent) {
-            console.log(`${i}: ${player_name || player_codes[player_num]} (${player_character}) beat ${opponent_name || opponent_code} (${opponent_character}) in ${game_length}!`)
-            total_wins++
-            total_games++
-            character_totals[player_character_num] = (character_totals[player_character_num] + 1) || 1
-            character_wins[player_character_num] = (character_wins[player_character_num] + 1) || 1
-            nickname_totals[player_name] = (nickname_totals[player_name] + 1) || 1
-            nickname_wins[player_name] = (nickname_wins[player_name] + 1) || 1
-            opponent_totals[opponent_code] = (opponent_totals[opponent_code] + 1) || 1
-            opponent_wins[opponent_code] = (opponent_wins[opponent_code] + 1) || 1
-
-        } else {
-            console.log(`${i}: ${player_name || player_codes[player_num]} (${player_character}) lost to ${opponent_name || opponent_code} (${opponent_character}) in ${game_length}.`)
-            total_games++
-            character_totals[player_character_num] = (character_totals[player_character_num] + 1) || 1
-            nickname_totals[player_name] = (nickname_totals[player_name] + 1) || 1
-            opponent_totals[opponent_code] = (opponent_totals[opponent_code] + 1) || 1
+        if (r.total_wins == 1) {
+            character_wins[r.player_character_num] = (character_wins[r.player_character_num] + 1) || 1
+            nickname_wins[r.player_name] = (nickname_wins[r.player_name] + 1) || 1
+            opponent_wins[r.opponent_code] = (opponent_wins[r.opponent_code] + 1) || 1
         }
+    })
+    var endTime = performance.now()
+    //console.log("Processing took " + (endTime - startTime) + " milliseconds.")
+    printReport()
```

Comment:
`printReport` needs to be invoked from within the `async function` otherwise it will execute instantly, before the results are ready.